lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1542,6 +1543,49 @@ SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
                                transValue(RV->getReturnValue(), F, BB), BB));
   }
 
+  case OpLifetimeStart: {
+    SPIRVLifetimeStart *LTStart = static_cast<SPIRVLifetimeStart*>(BV);
+    IRBuilder<> Builder(BB);
+    SPIRVWord Size = LTStart->getSize();
+	ConstantInt *S = nullptr;
+    if (Size) {
+      S = Builder.getInt64(Size);
+    }
+    Value* Var = transValue(LTStart->getObject(), F, BB);
+    CallInst *Start = Builder.CreateLifetimeStart(Var, S);
+    return mapValue(BV, Start->getOperand(1));
+  }
+
+  case OpLifetimeStop: {
+    SPIRVLifetimeStop *LTStop = static_cast<SPIRVLifetimeStop*>(BV);
+    IRBuilder<> Builder(BB);
+    SPIRVWord Size = LTStop->getSize();
+    ConstantInt *S = nullptr;
+    CallInst *Stop = nullptr;
+    if (Size)
+      S = Builder.getInt64(Size);
+    auto Var = transValue(LTStop->getObject(), F, BB);
+    for (const auto &I : Var->users()) {

```

Comment:
Can we refactor this part to a function getBitcastForLifeStart(X) ?

If it exists, returns the bitcast, otherwise X itself. Then just have one call of Builder.CreateLifetimeEnd.